pkg/loki/modules.go

```diff
@@ -287,6 +288,21 @@ func (t *Loki) initTableManager() error {
 	}
 
 	t.tableManager.Start()
+
+	t.server.HTTP.Path("/ready").Handler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// time.Sleep(60 * time.Seconds)
+		err := t.tableManager.SyncTables(context.Background())
```

Comment:
This is not a safe way to do. The loop here https://github.com/cortexproject/cortex/blob/master/pkg/chunk/table_manager.go#L190 always does `syncTables`, we might overlap the function call. 

Other way would be we can call `ListTables` and check if it gives some number without error. However, cortex doesn't give us a way to check if the returned number is same as expected number.

Any other better ways?

Reply:
Ready should only verify the process is still running, so you should use a ready we already have somewhere else returning only 200 and doing nothing else.

Reply:
see https://github.com/cortexproject/cortex/blob/fc47e09c1beb77d72ed5d86fdba99243498f8b7c/pkg/cortex/modules.go#L300

Reply:
Just for my clarity. `SyncTables` returns errors at different stage. So, if `SyncTables` hasn't gone through correctly then the table manager won't be able to do any action. We might say table-manager is ready since process is running but it might not be actually ready to do any action. Is this behaviour OK?

Reply:
We should only check if the sync manager is running properly. Unfortunately, cortex doesn't expose if the manager is still running although that should be easy to add reading their code:

```go
func (m *TableManager) IsRunning() bool {
     select {
      case <-m.done:
        return false
      default:
        return true
}
```

Then you can use that in your ready handler.

I'd suggest for now an empty handler like I've linked you. 

Then you can create an issue to improve:
-  first send a PR to cortex
-  come back here and add the check, if it's not running then it should fail.

Reply:
Cool. Thanks